### ISSUE TITLE
CMake Tools sets intelliSenseMode clang-x64 on ARM device unless "-arch arm" is explicitly added

### ISSUE THREAD

**Issue by TrevorMuraro** [microsoft/vscode-cmake-tools#1482]
Environment is remote SSH from a Mac into a Raspberry Pi 4, compiler is Clang 9.0.1.

main.cpp:
```cpp
#include <string>

int main()
{
   return 0;
}
```

CMakeLists.txt:

```cmake
cmake_minimum_required(VERSION 3.13)
set (CMAKE_CXX_EXTENSIONS OFF)
set(CMAKE_DISBLE_IN_SOURCE_BUILD ON)
set(CMAKE_CXX_COMPILER /usr/bin/clang++-9)
add_executable(TEST main.cpp)
```
 
Project compiles without issues.

### Expected:

CMake Tools provides intelliSenseMode `clang-arm` to C++ Tools.

### Apparent Behavior:

CMake Tools provides IntelliSenseMode `clang-x64` to C++ tools, and all standard library #includes get red squiggles.

I can avoid this by adding the line `target_compile_options(TEST PUBLIC -arch arm)` to CMakeLists.txt, but this is _only_ needed for the intelliSenseMode - compiling works fine without it, and compiling with the added line gets this warning:

```
[build] clang: warning: argument unused during compilation: '-arch arm' [-Wunused-command-line-argument]
```

### CMake Tools Log

```
[main] Safe constructing new CMakeTools instance
[variant] Constructing VariantManager
[main] Constructing new CMakeTools instance
[main] Starting CMakeTools second-phase init
[variant] Loaded new set of variants
[main] CMakeTools instance initialization complete.
[kit] Reading kits file /home/pi/.local/share/CMakeTools/cmake-tools-kits.json
[kit] Successfully loaded 5 kits from /home/pi/.local/share/CMakeTools/cmake-tools-kits.json
[kit] Not reading non-existent kits file: /home/pi/embedded/.vscode/cmake-kits.json
[extension] Configuring workspace on open file:///home/pi/embedded
[kitsController] Start selection of kits. Found 7 kits.
[kitsController] Opening kit selection QuickPick
[kitsController] User selected kit {"label":"Clang 9.0.1","description":"Using compilers: C = /usr/bin/clang-9, CXX = /usr/bin/clang++-9","kit":{"name":"Clang 9.0.1","compilers":{"C":"/usr/bin/clang-9","CXX":"/usr/bin/clang++-9"}}}
[main] Injecting new Kit into CMake driver
[main] Configuring folder: embedded 
[main] Saving open files before configure/build
[main] Starting new CMake driver
[main] Starting CMake driver
[main] Using generator from user configuration: Unix Makefiles
[driver] CMakeDriver Kit set to Clang 9.0.1
[driver] Run _refreshExpansions
[cms-driver] Run doRefreshExpansions
[driver] Run _refreshExpansions cb
[cms-client] Started new CMake Server instance with PID 22089
[cms-client] Configuring using the "Unix Makefiles" CMake generator
[driver] Setting new variant , Disable optimizations - include debug information.
[driver] Run _refreshExpansions
[cms-driver] Run doRefreshExpansions
[driver] Run _refreshExpansions cb
[cms-client] Started new CMake Server instance with PID 22108
[cms-client] Configuring using the "Unix Makefiles" CMake generator
[driver] Run _refreshExpansions
[cms-driver] Run doRefreshExpansions
[driver] Run _refreshExpansions cb
[driver] Start configure 
[driver] Runnnig pre-configure checks and steps
[driver] Using compilers in Clang 9.0.1 for configure
[driver] Run _refreshExpansions
[cms-driver] Run doRefreshExpansions
[driver] Run _refreshExpansions cb
[cmake] The C compiler identification is Clang 9.0.1
[cmake] The CXX compiler identification is Clang 9.0.1
[cmake] Check for working C compiler: /usr/bin/clang-9
[cmake] Check for working C compiler: /usr/bin/clang-9 -- works
[cmake] Detecting C compiler ABI info
[cmake] Detecting C compiler ABI info - done
[cmake] Detecting C compile features
[cmake] Detecting C compile features - done
[cmake] Check for working CXX compiler: /usr/bin/clang++-9
[cmake] Check for working CXX compiler: /usr/bin/clang++-9 -- works
[cmake] Detecting CXX compiler ABI info
[cmake] Detecting CXX compiler ABI info - done
[cmake] Detecting CXX compile features
[cmake] Detecting CXX compile features - done
[cmake] Configuring done
[cmake] Generating done
[cache] Reading CMake cache file /home/pi/embedded/build/CMakeCache.txt
[cache] Parsing CMake cache string
[cache] Reading CMake cache file /home/pi/embedded/build/CMakeCache.txt
[cache] Parsing CMake cache string
```

### Developer Tools Log

```
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] (node:76377) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
t.log @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287
workbench.html:1 A cookie associated with a cross-site resource at https://visualstudio.com/ was set without the `SameSite` attribute. A future release of Chrome will only deliver cookies with cross-site requests if they are set with `SameSite=None` and `Secure`. You can review cookies in developer tools under Application>Storage>Cookies and see more details at https://www.chromestatus.com/feature/5088147346030592 and https://www.chromestatus.com/feature/5633521622188032.
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:5411 No search provider registered for scheme: vscode-remote, waiting
(anonymous) @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:5411
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:14.316Z [debug] [main] Safe constructing new CMakeTools instance
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:14.323Z [debug] [variant] Constructing VariantManager
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:14.329Z [debug] [main] Constructing new CMakeTools instance
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:14.331Z [debug] [main] Starting CMakeTools second-phase init
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:15.096Z [info] [variant] Loaded new set of variants
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:15.107Z [debug] [main] CMakeTools instance initialization complete.
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:15.218Z [debug] [kit] Reading kits file /home/pi/.local/share/CMakeTools/cmake-tools-kits.json
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:15.369Z [info] [kit] Successfully loaded 5 kits from /home/pi/.local/share/CMakeTools/cmake-tools-kits.json
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:15.380Z [debug] [kit] Not reading non-existent kits file: /home/pi/embedded/.vscode/cmake-kits.json
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:15.452Z [debug] [extension] Configuring workspace on open file:///home/pi/embedded
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:15.457Z [debug] [kitsController] Start selection of kits. Found 7 kits.
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:15.458Z [debug] [kitsController] Opening kit selection QuickPick
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:16.905Z [debug] [kitsController] User selected kit {"label":"Clang 9.0.1","description":"Using compilers: C = /usr/bin/clang-9, CXX = /usr/bin/clang++-9","kit":{"name":"Clang 9.0.1","compilers":{"C":"/usr/bin/clang-9","CXX":"/usr/bin/clang++-9"}}}
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:16.913Z [debug] [main] Injecting new Kit into CMake driver
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:16.925Z [info] [main] Configuring folder: embedded 
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:16.928Z [debug] [main] Saving open files before configure/build
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:17.183Z [debug] [main] Starting new CMake driver
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:17.187Z [debug] [main] Starting CMake driver
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:17.189Z [debug] [main] Using generator from user configuration: Unix Makefiles
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:17.196Z [debug] [driver] CMakeDriver Kit set to Clang 9.0.1
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:17.202Z [debug] [driver] Run _refreshExpansions
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:17.204Z [debug] [cms-driver] Run doRefreshExpansions
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:17.206Z [debug] [driver] Run _refreshExpansions cb
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:17.307Z [debug] [cms-client] Started new CMake Server instance with PID 22089
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:18.322Z [info] [cms-client] Configuring using the "Unix Makefiles" CMake generator
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:18.339Z [debug] [driver] Setting new variant , Disable optimizations - include debug information.
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:18.341Z [debug] [driver] Run _refreshExpansions
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:18.342Z [debug] [cms-driver] Run doRefreshExpansions
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:18.344Z [debug] [driver] Run _refreshExpansions cb
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:18.422Z [debug] [cms-client] Started new CMake Server instance with PID 22108
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:19.429Z [info] [cms-client] Configuring using the "Unix Makefiles" CMake generator
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:20.090Z [debug] [driver] Run _refreshExpansions
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:20.091Z [debug] [cms-driver] Run doRefreshExpansions
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:20.092Z [debug] [driver] Run _refreshExpansions cb
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:20.124Z [debug] [driver] Start configure 
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:20.126Z [debug] [driver] Runnnig pre-configure checks and steps
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:20.137Z [debug] [driver] Using compilers in Clang 9.0.1 for configure
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:20.144Z [debug] [driver] Run _refreshExpansions
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:20.146Z [debug] [cms-driver] Run doRefreshExpansions
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:20.147Z [debug] [driver] Run _refreshExpansions cb
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:20.886Z [info] [cmake] The C compiler identification is Clang 9.0.1
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:21.791Z [info] [cmake] The CXX compiler identification is Clang 9.0.1
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:21.845Z [info] [cmake] Check for working C compiler: /usr/bin/clang-9
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:22.818Z [info] [cmake] Check for working C compiler: /usr/bin/clang-9 -- works
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:22.829Z [info] [cmake] Detecting C compiler ABI info
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:23.713Z [info] [cmake] Detecting C compiler ABI info - done
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:23.772Z [info] [cmake] Detecting C compile features
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:26.494Z [info] [cmake] Detecting C compile features - done
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:26.533Z [info] [cmake] Check for working CXX compiler: /usr/bin/clang++-9
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:27.768Z [info] [cmake] Check for working CXX compiler: /usr/bin/clang++-9 -- works
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:27.775Z [info] [cmake] Detecting CXX compiler ABI info
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:28.980Z [info] [cmake] Detecting CXX compiler ABI info - done
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:29.034Z [info] [cmake] Detecting CXX compile features
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:35.848Z [info] [cmake] Detecting CXX compile features - done
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:35.862Z [info] [cmake] Configuring done
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:35.886Z [info] [cmake] Generating done
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:36.087Z [debug] [cache] Reading CMake cache file /home/pi/embedded/build/CMakeCache.txt
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:36.095Z [debug] [cache] Parsing CMake cache string
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] activating extension
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] starting language server
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] (node:21984) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
t.log @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:37.170Z [debug] [cache] Reading CMake cache file /home/pi/embedded/build/CMakeCache.txt
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] [CMakeTools] 2020-10-11T06:11:37.395Z [debug] [cache] Parsing CMake cache string
2/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:287 [Extension Host] updateCustomBrowseConfiguration
2/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1707   ERR No application in the Launch Services database matches the input criteria.: Error: No application in the Launch Services database matches the input criteria.
DevTools failed to load SourceMap: Could not load content for file:///Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar/semver-umd/lib/semver-umd.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar/tas-client-umd/lib/tas-client-umd.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar/xterm/lib/xterm.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar/xterm-addon-unicode11/lib/xterm-addon-unicode11.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar/vscode-textmate/release/main.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar/xterm-addon-search/lib/xterm-addon-search.js.map: 
```

### Platform and Versions

- **Operating System**: Raspbian Buster, accessed via Remote SSH from macOS 10.15.7
- **CMake Version**: 3.13
- **VSCode Version**: 1.50.0
- **C/C++ Extension Version**: 1.0.1
- **CMake Tools Extension Version**: 1.4.2
- **Compiler/Toolchain**: Clang 9.0.1

### Other Notes/Information

For anyone else running into this, you can silence the "unused argument" warning from the workaround by adding the compiler option `-Qunused-arguments`. Older versions of Clang use something different, I believe `-Wno-error=unused-command-line-argument`.

**Comment by bobbrow** [comment]
Thank you for reporting this issue. We’ll let you know if we need more info to investigate it.

**Comment by bobbrow** [comment]
Is this fixed with version 1.7.1?  cpptools recently made it optional for configuration providers to set the `intelliSenseMode` since it already does work to figure this out, so we skip it unless an older version of cpptools is detected.
